### ISSUE TITLE
jax.random.multivariate_normal produces incorrect output with batches of covariance matrices

### ISSUE THREAD

**Issue by slinderman** [google/jax#3997]
The tensordot on [this line](https://github.com/google/jax/blob/fb02cc80f747bce8e01c4c80ca80f066945a8d3c/jax/random.py#L674) does not perform the intended broadcast matrix-vector multiply. I think you need this (or equivalent),
```
return mean + np.einsum('...ij,...j->...i', chol_factor, normal_samples)
```
Here's an example:
```
import jax.numpy as np
import jax.random

mean = np.zeros((10, 4))
cov = np.eye(4)[None,...].repeat(10, axis=0)
rng = jax.random.PRNGKey(0)
sample = jax.random.multivariate_normal(rng, mean, cov)
print(sample.shape) # is (10, 10, 4); should be (10, 4).
```

**Comment by jakevdp** [comment]
Thanks for the report – I agree that the einsum solution you propose looks correct. Would you like to submit a pull request with the fix?

**Comment by mattjj** [comment]
Yeah! Let's get @slinderman's first of many JAX PRs here!

**Comment by mattjj** [comment]
I left this bug in there just for you to find, Scott.

**Comment by slinderman** [comment]
Ha, thanks @mattjj! I found another minor bug in the shape checking.  Submitted a PR with a new test case here: https://github.com/google/jax/pull/4002.

**Comment by mattjj** [comment]
Merged the fix from Scott in #4002!
